src/UI/Components/Diagram/init.ts

```diff
@@ -124,239 +68,46 @@ export default function diagramInit(
-      document.dispatchEvent(
-        new CustomEvent("openDictsModal", {
-          detail: event.target.parentElement.attributes.dict.value,
-        }),
-      );
-    },
-  );
-
-  paper.on(
-    "element:toggleButton:pointerdown",
-    (elementView: dia.ElementView, event: dia.Event) => {
-      event.preventDefault();
-      const elementAsShape = elementView.model as ServiceEntityBlock;
+  //trigger highlighter when user drag element from stencil
+  graph.on("add", function (cell) {
+    const paperRepresentation = paper.findViewByModel(cell);
 
-      const isCollapsed = elementAsShape.get("isCollapsed");
-      const originalAttrs = elementAsShape.get("dataToDisplay");
-
-      elementAsShape.appendColumns(
-        isCollapsed ? originalAttrs : originalAttrs.slice(0, 4),
-        false,
-      );
-      elementAsShape.attr(
-        "toggleButton/xlink:href",
-        isCollapsed ? collapseButton : expandButton,
-      );
-
-      const bbox = elementAsShape.getBBox();
-
-      elementAsShape.attr("toggleButton/y", bbox.height - 24);
-      elementAsShape.attr("spacer/y", bbox.height - 33);
-      elementAsShape.attr("buttonBody/y", bbox.height - 32);
-
-      elementAsShape.set("isCollapsed", !isCollapsed);
-    },
-  );
-
-  paper.on("cell:pointerup", function (cellView) {
-    // We don't want a Halo if cellView is a Link or is a representation of an already existing instance that has strict_modifier set to false
     if (
-      cellView.model instanceof dia.Link ||
-      cellView.model.get("isBlockedFromEditing")
-    )
-      return;
-    if (cellView.model.get("isBlockedFromEditing") || !editable) return;
-    const halo = createHalo(
-      graph,
-      paper,
-      cellView,
-      connectionRules,
-      updateInstancesToSend,
-    );
-
-    halo.render();
-  });
-
-  paper.on("link:mouseenter", (linkView) => {
-    const source = linkView.model.source();
-    const target = linkView.model.target();
-
-    const sourceCell = graph.getCell(
-      source.id as dia.Cell.ID,
-    ) as ServiceEntityBlock;
-    const targetCell = graph.getCell(
-      target.id as dia.Cell.ID,
-    ) as ServiceEntityBlock;
-
-    if (!(sourceCell.getName()[0] === "_")) {
-      linkView.model.appendLabel({
-        attrs: {
-          rect: {
-            fill: "none",
-          },
-          text: {
-            text: sourceCell.getName(),
-            autoOrient: "target",
-            class: "joint-label-text",
-          },
-        },
-        position: {
-          distance: 1,
-        },
-      });
-    }
-    if (!(targetCell.getName()[0] === "_")) {
-      linkView.model.appendLabel({
-        attrs: {
-          rect: {
-            fill: "none",
-          },
-          text: {
-            text: targetCell.getName(),
-            autoOrient: "source",
-            class: "joint-label-text",
-          },
-        },
-        position: {
-          distance: 0,
-        },
-      });
+      cell.get("isEmbedded") &&
```

Comment:
For future iteration, I think the term isEmbedded and embeddedTo are meaning a bit the same, while they aren't meant to mean the same. I guess isEmbedded is supposed to be isEmbeddedEntity, and means that the type of entity is an embedded type, not that it is already embedded.

Reply:
https://github.com/inmanta/web-console/issues/6011